src/features/session_replay/instrument/index.js

```diff
@@ -16,23 +16,25 @@ export class Instrument extends InstrumentBase {
   static featureName = FEATURE_NAME
 
   #mode
-  constructor (agentIdentifier, aggregator, auto = true) {
-    super(agentIdentifier, aggregator, FEATURE_NAME, auto)
+  #agentRef
+  constructor (agentRef, auto = true) {
+    super(agentRef, FEATURE_NAME, auto)
     let session
     this.replayRunning = false
+    this.#agentRef = agentRef
```

Comment:
Should the inst base just set `this.agentRef` like the agg base does?

Reply:
So far nothing else has needed it outside the constructors since the instrument files are not complicated enough to warrant complex stack calls. I think we leave this until the cost arises as the change would be very easy at that stage.